adapters/criteo/criteo.go

```diff
@@ -64,30 +64,16 @@ func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.R
 
 	bidResponse := adapters.NewBidderResponse()
 	bidResponse.Currency = response.Cur
-	bidTypes := getMediaTypeForImps(request.Imp)
 
 	for _, seatBid := range response.SeatBid {
 		for i := range seatBid.Bid {
+
 			b := &adapters.TypedBid{
-				Bid:     &seatBid.Bid[i],
-				BidType: bidTypes[seatBid.Bid[i].ImpID],
```

Comment:
We cannot un-assign BidType. This has to be set else the response from your adapter will overwrite the `seatbid [i].bid.ext.prebid.type` value in the response. Refer to this comment on [TypedBid struct.](https://github.com/prebid/prebid-server/blob/6aab539f321ca50ffccf1d84f4399ec4567919ba/adapters/bidder.go#L94-L95)

I would suggest to set the BidType based on the `response.seatbid[i].bid.ext.prebid.type`.

Reply:
Hello, I have done the changes and I am now putting into BidType the value from `response.seatbid[i].bid.ext.prebid.type`